DFC.TestAutomation.UI/TestSupport/Configurator.cs

```diff
@@ -6,51 +6,43 @@
 
 namespace DFC.TestAutomation.UI.TestSupport
 {
-    internal static class Configurator
+    public static class Configurator
     {
-        private readonly static IConfigurationRoot _config;
+        private static IConfigurationRoot _config;
 
-        private readonly static IConfigurationRoot _hostingConfig;
+        private static IConfigurationRoot _hostingConfig;
 
-        internal readonly static bool IsVstsExecution;
+        public static bool IsVstsExecution { get; set; }
 
-        internal readonly static string EnvironmentName;
+        public static string EnvironmentName { get; set; }
 
-        internal readonly static string ProjectName;
+        public static string ProjectName { get; set; }
 
-        static Configurator()
-        {
-            _hostingConfig = InitializeHostingConfig();
-            IsVstsExecution = TestsExecutionInVsts();
-            EnvironmentName = GetEnvironmentName();
-            ProjectName = GetProjectName();
-            _config = InitializeConfig();
-        }
-
-        internal static IConfigurationRoot GetConfig() => _config;
+        public static IConfigurationRoot GetConfig() => _config;
 
-        private static IConfigurationRoot InitializeConfig()
+        public static IConfigurationRoot InitializeConfig(string[] settingsFiles)
```

Comment:
We have 2 public methods here:

1. InitializeConfig
2. InitializeHostingConfig

Both look the same. The difference between the two is not obvious. The top one returns an IConfigurationRoot which is an object used to hold the settings data. The other doesn't return anything. I find that inconsistent.

Also because we have removed the constructor (didn't know static classes had constructors?!) methods such as GetHostingConfigSection(name) could throw a NRE. I think we need to revise the impact of removing this code.

The other thing I'm not sure on is the parameter type string[]. My reason is, if I'm using this pack for the first time I'm left scratching my head wondering what the string is. Is it a file name? Or is it a path? If it is a path do I need to include the file extension? Is it relative?

Taking a wider look at this class I think we could do some tidying up. I can't think of a reason why this would need to be static for example. Maybe the test projects could initialise a class using an interface provided by the NuGet to set up their own config? I know this isn't a small tweak.

I don't mind going with this to get a working model then enhancing it later so happy to just keep the above as discussion points.

Reply:
Hi James, agree with everything.  It would be useful to get this in so we can try out this approach. Happy to discuss  the ongoing approach before making refinements. Thanks